lib/install/install.sh

```diff
@@ -155,16 +155,72 @@ bpkg_install () {
 
   echo
 
+  local build_output=""
+  local build=""
+  local json=""
+
   for pkg in "${pkgs[@]}"; do
     if test -d "$(bpkg_realpath "$pkg" 2>/dev/null)"; then
-      if ! (cd "$pkg" && ${BPKG_DEPS_EXEC}); then
-        return 1
-      fi
+      pushd . >/dev/null || return 1
+      cd "$pkg" || return 1
+
+      pwd
+      ${BPKG_DEPS_EXEC} || return 1
 
       did_fail=0
+
+      ## handle global install on a given local directory path
+      if (( needs_global )); then
+        if test -f "bpkg.json"; then
+          json="$(cat bpkg.json 2>/dev/null)"
+        elif test -f "package.json"; then
+          json="$(cat package.json 2>/dev/null)"
+        else
+          popd >/dev/null || return 1
+          bpkg_warn 'Unable to determine bpkg.json'
+          return 1
+        fi
+
+        if [ -z "$json" ]; then
+          popd >/dev/null || return 1
+          bpkg_warn 'Unable to determine bpkg.json'
+          return 1
+        fi
+
+        ## install bin if needed
```

Comment:
Should this section be extracted to a new function, along with its analog in `bpkg_install_from_remote()`?

https://github.com/bpkg/bpkg/blob/c7f5a467ec53e61d2344f5d1ab5766949c3bffdf/lib/install/install.sh#L388-L392